core/src/main/java/com/linecorp/armeria/internal/TransportType.java

```diff
@@ -0,0 +1,69 @@
+package com.linecorp.armeria.internal;
+
+import static com.linecorp.armeria.common.util.NativeLibraries.isEpollAvailable;
+import static com.linecorp.armeria.common.util.NativeLibraries.isKqueueAvailable;
+
+import java.util.concurrent.ThreadFactory;
+import java.util.function.Function;
+
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.ServerChannel;
+import io.netty.channel.epoll.EpollEventLoopGroup;
+import io.netty.channel.epoll.EpollServerSocketChannel;
+import io.netty.channel.kqueue.KQueueEventLoopGroup;
+import io.netty.channel.kqueue.KQueueServerSocketChannel;
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.channel.socket.nio.NioServerSocketChannel;
+
+/**
+ * Native transport types.
+ */
+public enum TransportType {
+    NIO, EPOLL, KQUEUE;
+
+    /**
+     * Returns the {@link Class} of available {@link ServerChannel}.
+     */
+    public static Class<? extends ServerChannel> serverChannel() {
```

Comment:
How about using instance to decide which behavior rather than using condition?
```
NIO(NioServerSocketChannel.class, (nThreads, factory) -> {
            return new NioEventLoopGroup(nThreads, factory);
        });
```
Then just `TransportType.detectTransportType().getChannel()` and `TransportType.detectTransportType()..evenLoopGroup(...)` ?